src/lib/common/sol-platform-linux-common.c

```diff
@@ -263,3 +307,581 @@ sol_platform_impl_get_os_version(char **version)
+        if (!mpoint) {
+            SOL_ERR("Could not allocate mount point memory");
+            goto err;
+        }
+
+        if (mbuf.mnt_fsname)
+            mpoint->dev = strdup(mbuf.mnt_fsname);
+
+        if (mbuf.mnt_dir)
+            mpoint->mpoint = strdup(mbuf.mnt_dir);
+
+        if (mbuf.mnt_type)
+            mpoint->fstype = strdup(mbuf.mnt_type);
+
+        sol_buffer_init(&custom_opts);
+        mpoint->flags = get_mountflags(mbuf.mnt_opts, &custom_opts, &noauto);
+        mpoint->noauto = noauto;
+
+        if (custom_opts.used)
+            mpoint->opts = sol_buffer_steal(&custom_opts, NULL);
+
+        sol_buffer_fini(&custom_opts);
+
+        if (sol_ptr_vector_append(vector, mpoint) < 0) {
+            SOL_ERR("Could not append mount point to vector");
+            goto err;
+        }
+    }
+    endmntent(tab);
+    return 0;
+
+err:
+    SOL_PTR_VECTOR_FOREACH_IDX (vector, mp, i) {
+        sol_mount_point_free(mp);
+    }
+    sol_ptr_vector_clear(vector);
+    return -ENOMEM;
+}
+
+int
+sol_platform_impl_get_mount_points(bool hplug_only, struct sol_ptr_vector *vector)
+{
+    struct sol_ptr_vector mtab, sol_mtab, delete;
+    struct sol_mount_point *mp, *mp2;
+    uint16_t i, j;
+    int err;
+
+    if (!hplug_only) {
+        return parse_mount_point_file(SYS_MTAB_FILE, vector);
+    }
+
+    sol_ptr_vector_init(&mtab);
+    err = parse_mount_point_file(SYS_MTAB_FILE, &mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_mtab);
+
+    sol_ptr_vector_init(&sol_mtab);
+    err = parse_mount_point_file(SOL_MTAB_FILE, &sol_mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_sol_mtab);
+
+    sol_ptr_vector_init(&delete);
+    SOL_PTR_VECTOR_FOREACH_IDX (&sol_mtab, mp, i) {
+        SOL_PTR_VECTOR_FOREACH_IDX (&mtab, mp2, j) {
+            if (streq(mp2->dev, mp->dev) && streq(mp2->mpoint, mp->mpoint)) {
+                sol_ptr_vector_append(vector, mp2);
+                sol_ptr_vector_remove(&mtab, mp2);
+            } else {
+                sol_ptr_vector_append(&delete, mp2);
+            }
+        }
+        sol_mount_point_free(mp);
+    }
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&delete, mp, i) {
+        sol_mount_point_free(mp);
+    }
+
+    sol_ptr_vector_clear(&delete);
+    sol_ptr_vector_clear(&sol_mtab);
+    sol_ptr_vector_clear(&mtab);
+
+    return 0;
+
+err_sol_mtab:
+    sol_ptr_vector_clear(&sol_mtab);
+err_mtab:
+    sol_ptr_vector_clear(&mtab);
+    return err;
+}
+
+static int
+sol_mtab_remove_entry(const char *mpoint, struct sol_buffer *output)
+{
+    int fd = 0;
+    struct sol_vector lines;
+    struct sol_str_slice content = SOL_STR_SLICE_EMPTY;
+    struct sol_str_slice *itr;
+    struct sol_file_reader *reader;
+    uint16_t idx;
+
+    fd = open(SOL_MTAB_FILE, O_RDONLY | O_CLOEXEC);
+    if (!fd) {
+        SOL_ERR("Could not open "SOL_MTAB_FILE " - %s",
+            sol_util_strerrora(errno));
+        return -errno;
+    }
+
+    reader = sol_file_reader_fd(fd);
```

Comment:
Is there the necessity to `open()` the file and pass to this function? Can't you just use the function that creates a reader from a file path?